### ISSUE TITLE
cc65: Error reporting is broken (again)

### ISSUE THREAD

**Issue by kugelfuhr** [cc65/cc65#2312]
Using the current version:

```C
#include <stdlib.h>
typedef int boolean;
#define NULL            0
void Func1()
{
}
extern boolean Func2();
```

Results in:

```
test.c:3: Error: Macro redefinition is not identical
test.c:7: Warning: Implicit 'int' is an obsolete feature
test.c:7: Error: ',' or ';' expected after top level declarator
2 errors and 1 warnings generated.
```

The error in line 3 is ok, but what the heck do the errors for line 7 mean? There's nothing wrong in this line.

The funny thing is, if line 3 is commented out:

```C
#include <stdlib.h>
typedef int boolean;
//#define NULL            0
void Func1()
{
}
extern boolean Func2();
```

the errors in line 7 are gone.

**Comment by kugelfuhr** [comment]
This does also work without `stdlib.h` and redefining the reserved macro NULL:

```C
typedef int boolean;
#define FOO     bar
#define FOO     baz
void Func1()
{
}
extern boolean Func2();
```

**Comment by acqn** [comment]
SSince cc65 currently does preprocessing during C declaration parsing, a preprocessor error would interfere with the error recovery of the parser. A separated preprocessor would be TODO in the future, but for now we may also implement "error category" eg. tokenizer/syntax/constraint errors to fix/enhance error recovery.

**Comment by kugelfuhr** [comment]
This seems to have been working some time ago. 2.18 says:

```
test.c(3): Error: Macro redefinition is not identical
```

which is what I would expect. I think 2.18 also did preprocessing during C declaration parsing.

So while it's not a really serious error, it's definitely a regression which has cost me some time to debug. In my case, the dreamed up error was several hundred lines away from the macro redefinition and I knew it had worked with older compiler versions.

**Comment by mrdudz** [comment]
fixed via #2314
